### PR TITLE
refactor: applied clang-tidy

### PR DESCRIPTION
**Pull request by soblin**
Signed-off-by: Mamoru Sobue <mamoru.sobue@tier4.jp>

## Description

I applied `clang-tidy` to the [prior PR](https://github.com/autowarefoundation/autoware_common/pull/114) and updated the code.

## Tests performed

I could not suppress this warning eventually, but I think this is C-specific matter.

```
 warning: do not use pointer arithmetic [cppcoreguidelines-pro-bounds-pointer-arithmetic]
  const std::string map_path = std::string(argv[1]);
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: tmp/lanelet2_extension/src/check_right_of_way.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 // NOLINTBEGIN(readability-identifier-naming)
+// NOLINTEND
```
Comment (kenji-miyake) [inline tmp/lanelet2_extension/src/check_right_of_way.cpp:16]: Why did you add this line? I guess it's not necessary.

```suggestion
```

File: tmp/lanelet2_extension/src/check_right_of_way.cpp [hunk 2]
```diff
@@ -51,15 +52,15 @@ int main(int argc, char ** argv)
...
     lanelet::traffic_rules::TrafficRulesFactory::create(
       lanelet::Locations::Germany, lanelet::Participants::Vehicle);
   lanelet::routing::RoutingGraphPtr routing_graph_ptr =
-    lanelet::routing::RoutingGraph::build(*map, *trafficRules);
+    lanelet::routing::RoutingGraph::build(*map, *traffic_rules);
 
   auto rows = map->regulatoryElementLayer  // filter elem whose Subtype is RightOfWay
               |
-              ranges::view::filter([](auto && elem) {
+              ranges::views::filter([](auto && elem) {
```
Comment (kenji-miyake) [inline tmp/lanelet2_extension/src/check_right_of_way.cpp:63]: Just out of curiosity, what was the warning of this part?

Comment (kenji-miyake) [reply tmp/lanelet2_extension/src/check_right_of_way.cpp:63]: Oh, it's this one.
https://github.com/autowarefoundation/autoware_common/pull/114#discussion_r1007863983

**Comment by soblin** [comment]
@mitsudome-r can you approve this PR?

**Comment by kenji-miyake** [comment]
@soblin Unfortunately, CI is still failing.
https://github.com/autowarefoundation/autoware_common/actions/runs/4529440040/jobs/7977270986?pr=128

**Comment by soblin** [comment]
@kenji-miyake There was no way to suppress clang-tidy error for `argv[1]` because `main(int argc, char **argv)` forces its usage. I added `NOLINT` block as a workaround, and the CI passed.

**Review by kenji-miyake — APPROVED** [review]

**Review by mitsudome-r — APPROVED** [review]
